### PR TITLE
enable wartremover only for compile time.

### LINKED ISSUE #6: Add wartremover
wartremover was in this code base at one point but we had it commented out.

We should put it back and add suppressions where necessary.

### PR DESCRIPTION
**Pull request by hqtan**
fixes #6 

Selectively turning off some warts to get compilation to run first. 

We can choose to turn on the disabled warts later..

I've tested to ensure that the proposed wartremover settings will get code to compile successfully, and not introduce more failed tests than the master branch currently has :sweat_smile:

### CODE REVIEW

**Comment by ssanj** [comment]
Hey @hqtan,

Could you please test this Wartremover configuration against the solution branch (`solutions-20200728`)?

I got the following errors:

```
[error] /Users/sanj/ziptemp/pr-reviews/solutions/applied-scala/src/main/scala/com/reagroup/appliedscala/urls/repositories/Http4sMetascoreRepository.scala:16:24: [wartremover:EitherProjectionPartial] RightProjection#get is disabled - use RightProjection#toOption instead
[error]     val omdbURI: Uri = uri"http://www.omdbapi.com/"
[error]                        ^
[error] /Users/sanj/ziptemp/pr-reviews/solutions/applied-scala/src/main/scala/com/reagroup/appliedscala/urls/repositories/PostgresqlRepository.scala:19:80: [wartremover:Option2Iterable] Implicit conversion from Option to Iterable is disabled - use Option#toList instead
[error]       case MovieRow(name, synopsis, _) => Movie(name, synopsis, rows.flatMap(_.review))
[error]                                                                                ^
[error] /Users/sanj/ziptemp/pr-reviews/solutions/applied-scala/src/main/scala/com/reagroup/appliedscala/urls/repositories/PostgresqlRepository.scala:36:87: [wartremover:Option2Iterable] Implicit conversion from Option to Iterable is disabled - use Option#toList instead
[error]       case ((name, synopsis), movieRows) => Movie(name, synopsis, movieRows.flatMap(_.review
```

Ideally we want the current solution branch to pass as is with all the warts we care about turned on.

**Comment by hqtan** [comment]
Ta for pointing that out @ssanj ! :bow: 
I've added a couple more warts to exclude in wartremover configs, and have confirmed that the code in `solutions-20200728` branch will now compile with the updated wartremover configs.

**Comment by wjlow** [comment]
Thanks @hqtan 😄

**Review by shaun-whitely — APPROVED** [review]

File: wartremover.sbt [hunk 1]
```diff
@@ -1 +1,11 @@
-//wartremoverErrors in (Compile, compile) ++= Warts.allBut(Wart.Nothing, Wart.Any, Wart.Recursion)
+wartremoverErrors in (Compile, compile) ++= Warts.allBut(
+  Wart.Nothing,
+  Wart.Any,
+  Wart.Recursion,
```
Comment (ssanj) [inline wartremover.sbt:4]: I think we can remove `Wart.Recursion`.

**Comment by ssanj** [comment]
One more change @hqtan and you're through! 👍

**Comment by hqtan** [comment]
:joy: 
I've made another commit to remove `Wart.Recursion`, @ssanj !

**Review by sanjivsahayamrea — APPROVED** [review]

**Review by ssanj — APPROVED** [review]

**Pull request merged**